crates/consensus/src/leader_selection.rs

```diff
@@ -66,20 +67,23 @@ impl WeightedRoundRobin {
         let mut proposer = self
             .validators
             .iter()
-            .max_by(|a, b| {
-                a.priority
-                    .partial_cmp(&b.priority)
-                    .unwrap_or(Ordering::Equal)
-            })
+            .max_by(Self::priority)
             .unwrap()
             .clone();
         proposer.priority -= total_voting_power as f64;
         let proposer_name = proposer.peer_id.clone();
+
         self.validators.replace(proposer);
 
         proposer_name
     }
 
+    fn priority(a: &&AngstromValidator, b: &&AngstromValidator) -> Ordering {
+        a.priority
+            .partial_cmp(&b.priority)
+            .unwrap_or(Ordering::Equal)
```

Comment:
yah it's fine ig if it can never be NaN - when i read it though i thought there 'could' be unexpected behavior because of the conditional unwrap. but if not it's all good